src/python/pants/backend/python/rules/python_create_binary.py

```diff
@@ -4,29 +4,29 @@
 from pants.backend.python.rules.pex import Pex
 from pants.backend.python.rules.pex_from_target_closure import CreatePexFromTargetClosure
 from pants.backend.python.targets.python_binary import PythonBinary
-from pants.build_graph.address import Address
 from pants.engine.addressable import Addresses
-from pants.engine.legacy.graph import HydratedTarget
 from pants.engine.legacy.structs import PythonBinaryAdaptor
 from pants.engine.rules import UnionRule, rule
 from pants.engine.selectors import Get
 from pants.rules.core.binary import BinaryTarget, CreatedBinary
-from pants.rules.core.strip_source_roots import SourceRootStrippedSources
+from pants.rules.core.strip_source_roots import SourceRootStrippedSources, StripTargetRequest
 
 
 @rule
 async def create_python_binary(python_binary_adaptor: PythonBinaryAdaptor) -> CreatedBinary:
     # TODO(#8420) This way of calculating the entry point works but is a bit hackish.
-    entry_point = None
     if hasattr(python_binary_adaptor, "entry_point"):
         entry_point = python_binary_adaptor.entry_point
     else:
         sources_snapshot = python_binary_adaptor.sources.snapshot
-        if len(sources_snapshot.files) == 1:
-            target = await Get[HydratedTarget](Address, python_binary_adaptor.address)
```

Comment:
This is a good example of why I put up this PR. This is unnecessary - we don't actually care about having a `HydratedTarget`. This makes the code less clear and has a (minor) performance hit.